packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,116 @@
-  const params: Buffer = abi.rawEncode(args, values).toString('hex');
+export default function encodeCall(name: string, types: string[] = [], values: any[] = []): string {
+  const methodId = abi.methodID(name, types).toString('hex');
+  const params = abi.rawEncode(types, values).toString('hex');
   return '0x' + methodId + params;
 }
+
+export function parseCallValues(types: string[] = [], rawValues: any[] = []): any[] | never {
+  if(types.length !== rawValues.length) throw new Error(ERROR_MESSAGE_PREFIX + '. Supplied number of types and values do not match.');
+  return _.zipWith(types, rawValues, parseTypeValuePair);
+}
+
+export function parseTypeValuePair(type: string, rawValue: any): any | never {
+  if(/^\(.*\)$/.test(type)) return parseTuple(type, rawValue);
+  if(/^[^\[]+\[.*\]$/.test(type)) return parseArray(type, rawValue);
+  if(type === 'address') return parseAddress(type, rawValue);
+  else if(type === 'bool') return parseBool(type, rawValue);
+  else if(type === 'string') return parseString(type, rawValue);
+  else if(type === 'function') return parseFunction(type, rawValue);
+  else if(type.startsWith('bytes')) return parseBytes(type, rawValue);
+  else if(type.startsWith('uint')) return parseNumber(type, rawValue, true, true);
+  else if(type.startsWith('int')) return parseNumber(type, rawValue, false, true);
+  else if(type.startsWith('ufixed')) return parseNumber(type, rawValue, true, false);
+  else if(type.startsWith('fixed')) return parseNumber(type, rawValue, false, false);
+  else throw new Error(ERROR_MESSAGE(type, rawValue) + '. Unsupported or invalid type.');
+}
+
+function parseTuple(type: string, rawValue: any): any | never {
+  if(typeof rawValue === 'string') rawValue = rawValue.split(',');
+  if(rawValue.length === 0) return [];
+  const types = type.replace(/[{()}]/g, '').split(','); // Remove the parenthesis and split the tuple into types.
+  if(types.length !== rawValue.length) throw new Error(ERROR_MESSAGE(type, rawValue) + '. Invalid tuple length.');
+  return _.zipWith(types, rawValue, (typeElement, rawValueElement) => parseTypeValuePair(typeElement, rawValueElement));
+}
+
+function parseArray(type: string, rawValue: any): any | never {
+  if(typeof rawValue === 'string') rawValue = rawValue.split(',');
+  if(rawValue.length === 0) return [];
+  const size = type.match(/(.*)\[(.*?)\]$/)[2]; // Find number between '[*]'.
+  if(size !== '' && parseInt(size, 10) !== rawValue.length) throw new Error(ERROR_MESSAGE(type, rawValue) + '. Invalid array length.');
+  const baseType = type.slice(0, type.lastIndexOf('[')); // Remove array part '[*]'.
+  return _.map(rawValue, (rawValueElement) => parseTypeValuePair(baseType, rawValueElement));
+}
+
+function parseFunction(type: string, rawValue: string): string | never {
+  if(typeof rawValue !== 'string') throw new Error(ERROR_MESSAGE(type, rawValue)); // Runtime type check.
+  if(!/^(0x)[0-9a-f]{46}$/i.test(rawValue)) throw new Error(ERROR_MESSAGE(type, rawValue));
```

Comment:
Why not fall back to `parseNumber` here, instead of forcing a hex value?

Reply:
Because I want to make sure that the value has exactly 24 bytes (an address + a function selector hash).